src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/MethodLdtokenNode.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Collections.Generic;
+
+using ILCompiler.DependencyAnalysisFramework;
+using Internal.Text;
+using Internal.TypeSystem;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// This method node is used for dependency tracking, like in the case of GVMs for example.
+    /// </summary>
+    internal class MethodLdtokenNode : DependencyNodeCore<NodeFactory>
+    {
+        private MethodDesc _method;
+
+        public MethodLdtokenNode(MethodDesc method)
+        {
+            Debug.Assert(!method.IsRuntimeDeterminedExactMethod);
+            _method = method;
+        }
+
+        public override bool HasConditionalStaticDependencies => false;
+        public override bool InterestingForDynamicDependencyAnalysis => false;
+        public override bool StaticDependenciesAreComputed => true;
+        protected override string GetName() => "__MethodLdtokenNode_" + NodeFactory.NameMangler.GetMangledMethodName(_method);
+
+        public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFactory context)
+        {
+            return Array.Empty<DependencyListEntry>();
+        }
+        public override IEnumerable<CombinedDependencyListEntry> GetConditionalStaticDependencies(NodeFactory context)
+        {
+            return Array.Empty<CombinedDependencyListEntry>();
+        }
+
+        public override bool HasDynamicDependencies
+        {
+            get
+            {
+                // Only generic virtual methods work have dynamic dependencies
+                if (_method.HasInstantiation && _method.IsVirtual)
+                {
+                    if (_method.IsCanonicalMethod(CanonicalFormKind.Specific))
+                        return false;
+
+                    if (_method.OwningType.ContainsGenericVariables || _method.ContainsGenericVariables)
```

Comment:
Can we make this a check for generic definition instead? Or does the code in this pull request actually generate things such as a `Foo` instantiated over `IFoo`'s `T` and the like that would warrant this expensive check?